src/metrics.jl

```diff
@@ -285,13 +284,13 @@ Base.eltype(d::PeriodicEuclidean) = eltype(d.periods)
     s3 = min(s2, p - s2)
     abs2(s3)
 end
+@inline function eval_op(d::PeriodicEuclidean, ai, bi)
+    periods = d.periods
+    p = isempty(periods) ? oneunit(eltype(periods)) : first(periods)
```

Comment:
Why is `oneunit(...)` appropriate here when the periods are empty? Is this tested?

Reply:
I'm not sure this function gets ever called? The empty test should be tested, I remember that I was struggling with that case at some point. I think, however, that in the empty periods case, `evaluate` will have checked before that `a` and `b` have also length zero, and that it doesn't make any difference what you put here for `p`. Apparently, I was doing something similar in the previous line 258 above.